pkg/ccr/job.go

```diff
@@ -2574,6 +2574,48 @@ func (j *Job) handleDropRollupRecord(commitSeq int64, dropRollup *record.DropRol
 	return j.IDest.DropRollup(destTableName, dropRollup.IndexName)
 }
 
+func (j *Job) handleRecoverInfo(binlog *festruct.TBinlog) error {
+	log.Infof("handle recoverInfo binlog, prevCommitSeq: %d, commitSeq: %d",
+		j.progress.PrevCommitSeq, j.progress.CommitSeq)
+
+	data := binlog.GetData()
+	recoverInfo, err := record.NewRecoverInfoFromJson(data)
+	if err != nil {
+		return err
+	}
+
+	return j.handleRecoverInfoRecord(binlog.GetCommitSeq(), recoverInfo)
+}
+
+func (j *Job) handleRecoverInfoRecord(commitSeq int64, recoverInfo *record.RecoverInfo) error {
+	if j.isBinlogCommitted(recoverInfo.TableId, commitSeq) {
+		return nil
+	}
+
+	if recoverInfo.PartitionName == "" {
```

Comment:
Consider adding a helper function `IsRecoverPartition() bool { return record.PartitionId == -1 }`, and change below line to:

```suggestion
	if recoverInfo.IsRecoverPartition() {
```